packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -748,11 +748,27 @@ export const Table = <
     ],
   );
 
+  const headerRef = React.useRef<HTMLDivElement>(null);
+  const bodyRef = React.useRef<HTMLDivElement>(null);
+
   const { scrollToIndex, tableRowRef } = useScrollToRow<T>({ ...props, page });
   const columnRefs = React.useRef<Record<string, HTMLDivElement>>({});
   const previousTableWidth = React.useRef(0);
   const onTableResize = React.useCallback(
     ({ width }: DOMRectReadOnly) => {
+      // ------------------------------------------------------------------------------------------------
+      // Handle header properties
+
+      setHeaderScrollWidth(headerRef.current?.scrollWidth ?? 0);
+      setHeaderClientWidth(headerRef.current?.clientWidth ?? 0);
+
+      // ------------------------------------------------------------------------------------------------
```

Comment:
nit: we typically use this divider comment for separating different components in the same file; it's not necessary here

Reply:
I prefer to keep them since I feel the two sections of the function need to be clearly divided. They are important as in the future, we should not accidentally add code to the wrong section. E.g. some column resizable logic added before the `if (!isResizable) { return; }`.

If not divided using these divider comments, I am open to other ways of making the division of the sections of the function clear.

Reply:
The actual words in the comments are fine. I'm talking only about the "----" comments, which are used for separating top-level components and functions (see `Alert.tsx` and `ListItem.tsx`). These separators do not help with maintainability; they are only there for improving readability.

If you want an actual way of improving maintainability, you could make a generic hook which returns these widths. There is already a `useContainerWidth` hook which you could modify.

Personally I'm ok with the code the way it is right now, other than the "----" comments. That's why this is marked as a nit.

Reply:
After deleting the "----" comments, I feel the sections are not clear. Because the word comment lines look just like standard comments and not section divider type comments.

So I reduced the length of the "----" comments (https://github.com/iTwin/iTwinUI/pull/1809/commits/ffd79209365db846d260216b379ef9f6e6bd8a3b).

But if you really don't want those "----" comments, we can go with [`e060df3` (#1809)](https://github.com/iTwin/iTwinUI/pull/1809/commits/e060df3b6badd640936572b568d860bb8b3799da) which removes those "----" comments and makes the text comments clearer.

Reply:
> After deleting the "----" comments, I feel the sections are not clear. Because the word comment lines look just like standard comments and not section divider type comments.

That's okay. Good, even. Like I said, these comments are best suited at the top-level scope.

When we refactor Table, we can write it in a way that avoids such long functions. This way we won't even need to have these discussions about comments.